app/javascript/components/Waybill.tsx

```diff
@@ -26,16 +26,19 @@ const Waybill = ({ currentUserRole }) => {
       componentMounted.current = false;
     };
   }, []);
-  const update_checkpoint_status = (id) => {
-    httpClient.route.get_routes(id).then((response) => setCheckpoints(response.data));
-
+  const update_checkpoint_status = (id, checkpoint) => {
+    const new_checkpoints = checkpoints;
+    const target_checkpoint = new_checkpoints.find((checkpoint) => checkpoint.id == id);
```

Comment:
Why do u use add var "new_checkpoins", do u can't use checkpoints arr directly? U need to use " === " here.